File: battleship-game/src/game/player.ts
import Ship from './ship';
import Board from './board';

export class Player {
    name: string;
    ships: Ship[];

    constructor(name: string) {
        this.name = name;
        this.ships = [];
    }

    makeMove(board: Board, coordinates: { x: number, y: number }): boolean {
        return board.checkHit(coordinates.x, coordinates.y);
    }

    checkWin(): boolean {
        return this.ships.every(ship => ship.isSunk());
    }

    addShip(ship: Ship): void {``
        this.ships.push(ship);
    }
}

File: src/game/player.py
from .ship import Ship
from .board import Board

class Player:
    def __init__(self, name):
        self.name = name
        self.ships = []

    def make_move(self, board, coordinates):
        return board.check_hit(coordinates[0], coordinates[1])

    def check_win(self):
        return all(ship.is_sunk() for ship in self.ships)

    def add_ship(self, ship):
        self.ships.append(ship)